ssl-config/src/main/java/com/palantir/remoting/ssl/KeyStores.java

```diff
@@ -0,0 +1,154 @@
+import java.security.GeneralSecurityException;
+import java.security.Key;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+
+final class KeyStores {
+
+    private KeyStores() {
+    }
+
+    /**
+     * Returns a {@link KeyStore} created by loading the certificate or certificates specified by the provided path.
+     *
+     * @param path a path to a X.509 certificate in PEM or DER format, or to a directory containing certificate files.
+     * If the path specifies a directory, every non-hidden file in the directory must be a X.509 certificate in PEM or
+     * DER format.
+     * @return a new KeyStore of type {@link KeyStore#getDefaultType()} that contains the certificates specified by the
+     * provided path. The returned store will not have any password.
+     */
+    static KeyStore createTrustStoreFromCertificates(Path path) {
+        try {
+            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            keyStore.load(null, null);
+
+            File pathFile = path.toFile();
+            if (pathFile.isDirectory()) {
+                File[] files = pathFile.listFiles(VISIBLE_FILE_FILTER);
+                if (files != null) {
+                    for (File currFile : files) {
+                        keyStore.setCertificateEntry(currFile.getName(), readX509Certificate(currFile.toPath()));
+                    }
+                }
+            } else {
+                keyStore.setCertificateEntry(pathFile.getName(), readX509Certificate(pathFile.toPath()));
+            }
+
+            return keyStore;
+        } catch (GeneralSecurityException | IOException e) {
+            throw Throwables.propagate(e);
+        }
+    }
+
+    /**
+     * Returns a {@link KeyStore} created by by loading all of the entries in the store specified by the provided
+     * parameters.
+     *
+     * @param storeType the type of the store specified by the provided path. Must be a type accepted by {@link
+     * KeyStore#getInstance(String)}. The returned store will also be of this type.
+     * @param path the path to the existing store to load.
+     * @param password an optional password used to read the provided store. If absent, no password will be used to read
+     * the store.
+     * @return a new KeyStore of type storeType that contains the entries read from the store at the provided path using
+     * the provided password.
+     */
+    static KeyStore loadKeyStore(String storeType, Path path, Optional<String> password) {
+        try {
+            KeyStore keyStore = KeyStore.getInstance(storeType);
+            try (InputStream stream = Files.newInputStream(path)) {
+                char[] passwordChars = password.isPresent() ? password.get().toCharArray() : null;
+                keyStore.load(stream, passwordChars);
+            }
+            return keyStore;
+        } catch (GeneralSecurityException | IOException e) {
```

Comment:
This is called from `SslSocketFactories`, so code ends up being a bit cleaner by doing the try/catch here (could make exceptions checked and make the entire calling block in `SslSocketFactories` catch exception, but don't see much of an upside to making the change)